network/eos/eos_command.py

```diff
@@ -109,46 +143,74 @@
   sample: ['...', '...']
 """
 from ansible.module_utils.basic import get_exception
-from ansible.module_utils.netcmd import CommandRunner, FailedConditionsError
-from ansible.module_utils.network import NetworkError
-from ansible.module_utils.eos import get_module
+from ansible.module_utils.netcli import CommandRunner
+from ansible.module_utils.netcli import AddCommandError, FailedConditionsError
+from ansible.module_utils.eos import NetworkModule, NetworkError
+
+VALID_KEYS = ['command', 'output', 'prompt', 'response']
 
 def to_lines(stdout):
     for item in stdout:
         if isinstance(item, basestring):
             item = str(item).split('\n')
         yield item
 
+def parse_commands(module):
+    for cmd in module.params['commands']:
+        if isinstance(cmd, basestring):
+            cmd = dict(command=cmd, output=None)
+        elif 'command' not in cmd:
+            module.fail_json(msg='command keyword argument is required')
+        elif cmd.get('output') not in [None, 'text', 'json']:
+            module.fail_json(msg='invalid output specified for command')
+        elif not set(cmd.keys()).issubset(VALID_KEYS):
+            module.fail_json(msg='unknown keyword specified')
```

Comment:
as a side note, we should port this change over to the other network *_command modules